### ISSUE TITLE
Authentication with OIDC.

### ISSUE THREAD

**Issue by RiddlerKnight** [boltops-tools/terraspace_plugin_azurerm#21]
On the CI/CD, I want to reduce using of secrets key. I think this feature is crucial.

azurerm provider is also supported.
https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/service_principal_oidc#oidc-token

**Comment by RiddlerKnight** [comment]
I Just tested and work by using this env on pipeline.
```
  ARM_CLIENT_ID: ""
  ARM_TENANT_ID: ""
  ARM_SUBSCRIPTION_ID: ""
  ARM_USE_CLI: "false"
  ARM_USE_OIDC: "true"
  ARM_OIDC_TOKEN: ""
```

**Comment by HoLengZai** [comment]
This issue should not be closed because your "solution" is to use Terraform env var to handle the OIDC authentication so the terraspace_plugin_azurerm does not perform anything.
So i made a PR to fix this issue which is not directly related to terraspace_plugin_azurerm
[https://github.com/boltops-tools/armrest/pull/9](https://github.com/boltops-tools/armrest/pull/9)

Without the PR above from armrest gem, terraspace will fail to auto create the tf backend because the plugin cannot authenticate with OIDC

**Comment by tongueroo** [comment]
Fixed by https://github.com/boltops-tools/armrest/pull/9 Released in armrest 0.2.3  

For those who are not as familiar with Ruby. You can run this in your terraspace project (that has the Gemfile)

    bundle update armrest

Or to update all (which is good long term)

    bundle update
